api/v1alpha1/protectedapplicationview_types.go

```diff
@@ -0,0 +1,165 @@
+package v1alpha1
+
+import (
+	ramenv1alpha1 "github.com/ramendr/ramen/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type ApplicationType string
+
+const (
+	// ApplicationSetType represents an ArgoCD ApplicationSet application
+	ApplicationSetType ApplicationType = "ApplicationSet"
+
+	// SubscriptionType represents an OpenShift Subscription application
+	SubscriptionType ApplicationType = "Subscription"
+
+	// DiscoveredType represents a discovered application (neither ApplicationSet nor Subscription)
+	DiscoveredType ApplicationType = "Discovered"
+)
+
+// ProtectedApplicationViewSpec defines the desired state of ProtectedApplicationView
+type ProtectedApplicationViewSpec struct {
+
+	// DRPCRef references the DRPlacementControl protecting this application
+	// +kubebuilder:validation:Required
+	DRPCRef corev1.ObjectReference `json:"drpcRef"`
+
+	// ApplicationRef references the source application resource
+	// +kubebuilder:validation:Required
+	ApplicationRef corev1.ObjectReference `json:"applicationRef"`
```

Comment:
We can take care of the ApplicationRef information from the DRPC. I don't think we need anything more in the spec